frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Model.java

```diff
@@ -109,19 +118,59 @@ public void setMaxBatchDelay(int maxBatchDelay) {
         this.maxBatchDelay = maxBatchDelay;
     }
 
+    public boolean addJob(String threadId, Job job) {
+        jobsDb.putIfAbsent(threadId, new LinkedBlockingDeque<>());
+        return jobsDb.get(threadId).offer(job);
+    }
+
+    public void removeJobQueue(String threadId) {
+        if (!threadId.equals(DEFAULT_DATA_QUEUE)) {
+            jobsDb.remove(threadId);
+        }
+    }
+
     public boolean addJob(Job job) {
-        return jobs.offer(job);
+        return addJob(DEFAULT_DATA_QUEUE, job);
+    }
+
+    public void addFirst(String threadId, Job job) {
+        jobsDb.putIfAbsent(threadId, new LinkedBlockingDeque<>());
+        jobsDb.get(threadId).addFirst(job);
     }
 
     public void addFirst(Job j) {
-        jobs.addFirst(j);
+        addFirst(DEFAULT_DATA_QUEUE, j);
+    }
+
+    public Job nextJob(String threadId) throws InterruptedException {
+        String tid = threadId != null ? threadId : DEFAULT_DATA_QUEUE;
```

Comment:
It seems threadId is never null. There is no call path passing null threadId.
This can be simplified as:

    public Job nextJob(String threadId) throws InterruptedException {
        return nextJob(threadId, Long.MAX_VALUE);
    }

    public Job nextJob(String threadId, long timeout) throws InterruptedException {
        LinkedBlockingDeque<Job> jobs = jobsDb.get(threadId);
        if (jobs != null && !jobs.isEmpty()) {
            return jobs.poll();
        }
        return jobsDb.get(DEFAULT_DATA_QUEUE).poll(timeout, TimeUnit.MILLISECONDS);
    }

Reply:
Sure. This looks more compact.